pkg/sink/cloudstorage/path.go

```diff
@@ -207,3 +188,58 @@ func (f *FilePathGenerator) GenerateIndexFilePath(tbl VersionedTable, date strin
 
 	return strings.Join(elems, "/")
 }
+
+func (f *FilePathGenerator) getNextFileIdxFromIndexFile(
+	ctx context.Context, tbl VersionedTable, date string,
+) (fileIdx uint64, err error) {
+	indexFile := f.GenerateIndexFilePath(tbl, date)
+	exist, err := f.storage.FileExists(ctx, indexFile)
+	if err != nil {
+		return fileIdx, err
+	}
+	if exist {
```

Comment:
May be we can do this first:
```go
if !exist {
  return 0,nil
}
```
So that the complicated logic below can move outside the if block.